corehq/messaging/scheduling/views.py

```diff
@@ -1061,3 +1065,20 @@ def post(self, request, *args, **kwargs):
             msg[0](request, msg[1])
 
         return self.get(request, *args, **kwargs)
+
+
+class CountCasesBasedOnCaseTypeView(APIView):
```

Comment:
Did you check if such a view already exists? Seems a common use case that should need to be reported on HQ. I see `get_number_of_cases_in_domain_of_type` but not sure if that is what you need.
Also, can we add authentication to this view?
You would definitely find utils method for this.

Reply:
Looks like the method doesn't exist for this particular use case `get_number_of_cases_in_domain_of_type` isn't what we need here. Pushed the authentication code.

Reply:
@mkangia Do you think the scheduling app is a the right place for this view? And do you know if we have the ability to authenticate both login and domain for API calls?

Reply:
How about `corehq.form_processor.interfaces.dbaccessors` for the util method at least? The view can then call it.

> And do you know if we have the ability to authenticate both login and domain for API calls
oh we don't? I am not sure. That data should be authenticated.